realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -75,41 +70,58 @@
  * A RealmObject cannot be passed between different threads.
  *
  * @see Realm#createObject(Class)
- * @see Realm#copyToRealm(RealmObject)
+ * @see Realm#copyToRealm(RealmModel)
  */
 
-@RealmClass
-public abstract class RealmObject {
+public abstract class RealmObject implements RealmModel {
 
-    protected Row row;
-    protected BaseRealm realm;
 
-    private final List<RealmChangeListener> listeners = new CopyOnWriteArrayList<RealmChangeListener>();
-    private Future<Long> pendingQuery;
-    private boolean isCompleted = false;
-    protected long currentTableVersion = -1;
+// TODO Move this to the proxy classes and expose methods in the RealmObjectProxy interface
```

Comment:
Is this still a TODO or an old comment? I saw the fields in the `RealmObjectProxy` below.